gulpfile.js

```diff
@@ -1102,6 +1103,28 @@ gulp.task("jsdoc", function(done) {
   });
 });
 
+gulp.task("dts", function(done) {
+  console.log();
+  console.log("### Generating typescript definitions");
+
+  var JSDOC_FILES = [
```

Comment:
This changed in the `jsdoc` target since this patch was made I think, so making this equal to https://github.com/mozilla/pdf.js/pull/10575/files#diff-b9e12334e9eafd8341a6107dd98510c9R1092 allows you to remove all `src/core` and `src/shared` changes from this patch and include all TypeScript-specific constructs in `src/doc_helper.js` as indicated by @Snuffleupagus.